app/controllers/searches_api_controller.rb

```diff
@@ -10,9 +10,10 @@ class SearchesApiController < ActionController::API
 
   # Show information about a registered search.
   def show
-    search = Search.find_by!(search_id: params[:id])
-    search.be_referred!
-    render pretty_json: search, except: :id
+    Search.find_by!(search_id: params[:id]).be_referred!
+
+    search = Search.dialog_search(params[:id])
```

Comment:
こんな感じにすると検索が一回で済みそうです。

```suggestion
    search = Search.includes(pseudo_graph_pattern: [:all_answers])
                    .find_by(search_id: search_id)
    search.be_referred!
    search = search.hash_dialog_search
```

検索するメソッドはSearchモデルのいい感じのメソッドにしておいてください。

Reply:
Searchモデルにdialog_searchメソッドに移動で対応しました。